plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduClientSession.java

```diff
@@ -639,6 +641,22 @@ private KuduSplit toKuduSplit(KuduTableHandle tableHandle, KuduScanToken token,
         }
     }
 
+    private String toRawTableName(SchemaTableName schemaTableName)
+    {
+        String tableName = schemaEmulation.toRawName(schemaTableName);
+        verify(tableName.equals(tableName.toLowerCase(ENGLISH)), "tableName not in lower-case: %s", tableName);
+        if (!caseInsensitiveNameMatching) {
+            return tableName;
+        }
+        String prefix = schemaEmulation.getPrefixForTablesOfSchema(schemaTableName.getSchemaName());
+        for (String remoteTableName : internalListTables(prefix)) {
```

Comment:
add caching for this functionality otherwise analysis will become very very slow. We had the issue in BigQuery.